combine resource and resource operations classes

m-nash opened the pull request:
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

m-nash committed:
update simple resources to no longer have operations
rename ResourceOperations to ArmResource
rename ResourceContainer to ArmContainer

m-nash committed:
combine genericResource and genericResourceExpanded
add test case to validate the expand deserializes properly

m-nash committed:
combine managementGroupInfo and manageGroup

allenjzhang commented on sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs:
```diff
@@ -157,40 +157,50 @@ public virtual TenantContainer GetTenants()
         /// <param name="id"> The id of the subscription. </param>
         /// <returns> Resource operations of the subscription. </returns>
-        public virtual SubscriptionOperations GetSubscriptionOperations(string id)
+        public virtual Subscription GetSubscription(string id)
```
These first class Get calls all taking string id as parameter. The expectation may not be clear on the format. For example,  for RG, you would expected the resource id as "sub/id/rg/id" points to the RG. But for subscription, providers etc, just the {id} rather than "/subscription/id" should just work. I suggest we do some string value check and support both cases.

m-nash replied:
maybe change the type to ResourceIdentifier to make it clear we expect a full id in all cases?

allenjzhang replied:
I would suggest for all non first level resources, the parameter to be ResourceIdentifier. For top-level resources, we can keep it as a string and be more lenient, so it is more convenient to use.

allenjzhang commented on sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs:
```diff
@@ -106,7 +106,7 @@ public ArmClient(TokenCredential credential, ArmClientOptions options)
+            _tenant = new Tenant(ClientOptions, Credential, BaseUri, Pipeline);
             DefaultSubscription = string.IsNullOrWhiteSpace(defaultSubscriptionId)
                 ? GetDefaultSubscription()
                 : GetSubscriptions().GetIfExists(defaultSubscriptionId);
```
GetIfExists's code seems to can return a null value but in reality, would have gotten 404 exception which is fine. Can we have a test case for an invalid subscription id?

m-nash committed:
update core tests after changes

m-nash committed:
update fields to be consistent with autorest

m-nash committed:
update after test interceptor changes

m-nash committed:
update to take resourceidentifier
differentiate between a tenant provider and a subscription provider
add overloads to get generic resource list to take string and resource identifier

m-nash committed:
update api after changes

m-nash committed:
add test case to validate bad sub id being passed into armclient

m-nash committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into mnash-6113-removeOperations

m-nash committed:
temporarily use alpha version of core

m-nash committed:
remove unused overload

m-nash committed:
add recording so test can be played back in ci

m-nash committed:
use testenv creds

ShivangiReja commented on sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs:
```diff
@@ -94,7 +91,7 @@ internal ResourceOperations(ClientContext clientContext, ResourceIdentifier id)
         /// </summary>
         /// <returns> A TagResourceOperations. </returns>
-        protected internal TagResourceOperations TagResourceOperations => _tagResourceOperations ??= new TagResourceOperations(this, Id);
+        protected internal TagResource TagResourceOperations => _tagResourceOperations ??= new TagResource(this, Id);
```
Can we update TagResourceOperations property name to TagResource?
